src/components/tasks/card/TaskStatusEditMode.tsx

```diff
@@ -1,42 +1,76 @@
 import { BACKEND_TASK_STATUS } from '@/constants/task-status';
-import task from '@/interfaces/task.type';
+import task, { CardTaskDetails } from '@/interfaces/task.type';
+import { useState } from 'react';
+import { toast, ToastTypes } from '@/helperFunctions/toast';
+import { SmallSpinner } from './SmallSpinner';
+import { SavedCheckmark } from './SavedCheckmark';
+import classNames from '@/components/tasks/card/card.module.scss';
+import { PENDING, SAVED } from '../constants';
+import { useUpdateTaskMutation } from '@/app/services/tasksApi';
 
 type Props = {
     task: task;
-    updateTask: (changeId: string, changeObject: { status: string }) => void;
+    setEditedTaskDetails: React.Dispatch<React.SetStateAction<CardTaskDetails>>;
 };
 
 // TODO: remove this after fixing the card beautify status
 const beautifyStatus = (status: string) => status.split('_').join(' ');
 const taskStatus = Object.entries(BACKEND_TASK_STATUS);
 
-const TaskStatusEditMode = ({ task, updateTask }: Props) => {
+const TaskStatusEditMode = ({ task, setEditedTaskDetails }: Props) => {
+    const { ERROR } = ToastTypes;
+    const [saveStatus, setSaveStatus] = useState('');
+    const [updateTask] = useUpdateTaskMutation();
+
     const onChangeUpdateTaskStatus = ({
         target: { value },
     }: React.ChangeEvent<HTMLSelectElement>) => {
-        updateTask(task.id, {
-            status: value,
+        setSaveStatus(PENDING);
+        setEditedTaskDetails((prev: any) => ({ ...prev, status: value }));
+        const response = updateTask({
+            id: task.id,
+            task: {
+                status: value,
+            },
         });
+
+        response
+            .unwrap()
+            .then((result: any) => {
+                setSaveStatus(SAVED);
+            })
+            .catch((err: { data: { message: string } }) => {
+                toast(ERROR, err.data.message);
+            })
+            .finally(() => {
+                setTimeout(() => {
+                    setSaveStatus('');
+                }, 3000);
+            });
     };
 
     // TODO: remove this after fixing the card beautify status
     const defaultStatus = task.status.toUpperCase().split(' ').join('_');
```

Comment:
in he defaultStatus variable is supposed to be the default value for the status select element. However, the code is converting the task's status to uppercase and then splitting it on spaces. This means that if the task's status is "In Progress", the defaultStatus variable will be set to "IN_PROGRESS". This is not the correct value for the status select element.